src/megaclient.cpp

```diff
@@ -1915,18 +1913,69 @@ void MegaClient::exec()
         }
 
         // handle API server-client requests
-        if (!jsonsc.pos && pendingsc && !loggingout)
+        if (pendingscUserAlerts)
+        {
+            switch (pendingscUserAlerts->status)
+            {
+            case REQ_SUCCESS:
+                if (*pendingscUserAlerts->in.c_str() == '{')
+                {
+                    JSON json;
+                    json.begin(pendingscUserAlerts->in.c_str());
+                    json.enterobject();
+                    if (useralerts.procsc_useralert(json))
+                    {
+                        // NULL vector: "notify all elements"
+                        app->useralerts_updated(NULL, int(useralerts.alerts.size()));
+                    }
+                    pendingscUserAlerts.reset();
+                    break;
+                }
+
+                // fall through
+            case REQ_FAILURE:
+                if (pendingscUserAlerts->httpstatus != 200)
+                {
+                    (pendingscUserAlerts->httpstatus == 500 ? fnstats.e500Count : fnstats.eOthersCount)++;
```

Comment:
This code is to count errors for the fetchnodes statistics. IIRC, the user-alerts are requested after the fetchnodes completes, so we should not update `fnstats` in result of errors getting the useralerts (well, it should be harmless, but useless as well, since the stats have been sent already).

Same for `fnstats.eAgainCount++;` in line 1946.

Reply:
Good to know, I was just keeping that from the `wsc` handling but in that case we don't need it, I've removed those lines, thanks